apis/generators/v1alpha1/generator_acr.go

```diff
@@ -0,0 +1,122 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/external-secrets/external-secrets/apis/externalsecrets/v1beta1"
+	smmeta "github.com/external-secrets/external-secrets/apis/meta/v1"
+)
+
+// ACRAccessTokenSpec defines how to generate the access token
+// e.g. how to authenticate and which registry to use.
+// see: https://github.com/Azure/acr/blob/main/docs/AAD-OAuth.md#overview
+type ACRAccessTokenSpec struct {
+	Auth ACRAuth `json:"auth"`
+	// TenantID configures the Azure Tenant to send requests to. Required for ServicePrincipal auth type.
+	TenantID string `json:"tenantId,omitempty"`
+
+	// the domain name of the ACR registry
+	// e.g. foobarexample.azurecr.io
+	ACRRegistry string `json:"registry"`
+
+	// Define the scope for the access token, e.g. pull/push access for a repository.
+	// if not provided it will return a refresh token that has full scope.
+	// Note: you need to pin it down to the repository level, there is no wildcard available.
+	//
+	// examples:
+	// repository:my-repository:pull,push
+	// repository:my-repository:pull
+	//
+	// see docs for details: https://docs.docker.com/registry/spec/auth/scope/
+	// +optional
+	Scope string `json:"scope,omitempty"`
+
+	// EnvironmentType specifies the Azure cloud environment endpoints to use for
+	// connecting and authenticating with Azure. By default it points to the public cloud AAD endpoint.
+	// The following endpoints are available, also see here: https://github.com/Azure/go-autorest/blob/main/autorest/azure/environments.go#L152
+	// PublicCloud, USGovernmentCloud, ChinaCloud, GermanCloud
+	// +kubebuilder:default=PublicCloud
+	EnvironmentType v1beta1.AzureEnvironmentType `json:"environmentType,omitempty"`
+}
+
+type ACRAuth struct {
```

Comment:
Is there any reason why the `ACRAuth` does not share the same SecretStore types as `ECR` and `GCR` does with AWS/GCP Secret Stores? Is it a different mechanism?

IMO it might be worth to use the standard from this one to the others (having the auth credentials defined in the generator), as it means we would be able to change them without coupling the code - I think it might be too soon to bind them together 😅 (as SecretStores are v1beta1, so changes there would be more difficult to go handle)